src/evict/evict_lru.c

```diff
@@ -448,8 +454,22 @@ __evict_server(WT_SESSION_IMPL *session, bool *did_work)
         return (0);
 
     __wt_epoch(session, &now);
-    if (WT_TIMEDIFF_SEC(now, cache->stuck_time) > WT_MINUTE * 5) {
-#if defined(HAVE_DIAGNOSTIC)
+
+#define WT_CACHE_STUCK_TIMEOUT_S 300
+#ifdef HAVE_DIAGNOSTIC
+    /* Enable extra logs before timing out. */
+    if (!verbose_timeout_flags &&
+      WT_TIMEDIFF_MS(now, cache->stuck_time) >
```

Comment:
I suggest adding a comment about the interesting calculation so that the reader doesn't have to really dig into that. Something like "Enable extra logging for the last 20 msecs before timing out."

Reply:
Sure, added in a09ffd452cb4ecc33171291465fb2d8f992b60fc